### PR TITLE
feat: Add static getter for TableDefinition in KafkaTools

### PR DESCRIPTION
**Pull request by robbcamera**

### CODE REVIEW

**Comment by robbcamera** [comment]
I have read the CLA Document and I hereby sign the CLA

File: extensions/kafka/src/main/java/io/deephaven/kafka/KafkaTools.java [hunk 1]
```diff
@@ -1297,12 +1294,61 @@ public static void consume(
...
+    /**
+     * Construct a {@link TableDefinition} based on the input Properties and {@link Consume.KeyOrValueSpec} parameters.
+     *
+     * @param kafkaProperties Properties to configure this table
+     * @param keySpec Conversion specification for Kafka record keys
+     * @param valueSpec Conversion specification for Kafka record values
+     * @return A TableDefinition derived from the input Properties and KeyOrValueSpec instances
+     */
+    @SuppressWarnings("unused")
+    public static TableDefinition getTableDefinition(
```
Comment (devinrsmith) [inline extensions/kafka/src/main/java/io/deephaven/kafka/KafkaTools.java:1309]: We should explicitly call out the potential statefulness of `KeyOrValueSpec`. If the caller is going to rely on the correctness of this method, they _must_ call consume with the exact same instances of keySpec / valueSpec.

ie, the following is _incorrect_:

```java
TableDefinition tableDefinition = getTableDefinition(properties, null, avroSchema("MyValueSchema"));
Table table = consume(properties, ..., avroSchema("MyValueSchema"), ...)
```

it must be

```java
KeyOrValueSpec myValueSpec = avroSchema("MyValueSchema");
TableDefinition tableDefinition = getTableDefinition(properties, null, myValueSpec);
Table table = consume(properties, ..., myValueSpec, ...)
```

Comment (devinrsmith) [reply extensions/kafka/src/main/java/io/deephaven/kafka/KafkaTools.java:1309]: Or gather, we should add documentation about the "guarantees" between getTableDefinition and consume; in what conditions is the table definition from getTableDefinition guaranteed to be the table definition of consume? (In my mind, it should be guaranteed when you use the exact same keySpec / valueSpec due to the statefulness of them.)

**Review by devinrsmith — APPROVED** [review]

**Comment by deephaven-internal** [comment]
Labels indicate documentation is required. Issues for documentation have been opened:

Community: https://github.com/deephaven/deephaven-docs-community/issues/300

**Pull request merged**